swagger_parser/lib/src/generator/templates/dart_enum_dto_template.dart

```diff
@@ -31,17 +31,28 @@ String dartEnumDtoTemplate(
     final unknownEnumValueStr = unknownEnumValue ? _unkownEnumValue() : '';
     final constructorStr = jsonParam ? _constructor(className) : '';
     final fromJsonStr = unknownEnumValue ? _fromJson(className, enumClass) : '';
-    final jsonFieldStr = jsonParam ? _jsonField(enumClass) : '';
-    final toJsonStr = enumsToJson ? _toJson(enumClass, className) : '';
+    final jsonFieldStr =
+        jsonParam ? _jsonField(enumClass, unknownEnumValue) : '';
+    final toJsonStr =
+        enumsToJson ? _toJson(enumClass, className, unknownEnumValue) : '';
+    final toStringStr = _toString(enumClass);
+    final compareToStr = _compareTo(className, enumClass, unknownEnumValue);
 
     return '''
 ${generatedFileComment(
       markFileAsGenerated: markFileAsGenerated,
     )}${dartImportDtoTemplate(jsonSerializer)}
 
 ${descriptionComment(enumClass.description)}@JsonEnum()
-enum $className {
-$values$unknownEnumValueStr$constructorStr$fromJsonStr$jsonFieldStr$toJsonStr
+enum $className implements Comparable<$className> {
+$values
+$unknownEnumValueStr
+$constructorStr
+$fromJsonStr
+$jsonFieldStr
+$toJsonStr
+$toStringStr
+$compareToStr
```

Comment:
I'm very unsure if you should override toString and add compareTo.

toString works correctly by default, if you only need the name of the enum, use .name

The compareTo implementation may only work in a rare special case. At the moment, it doesn't work for enum that have type string. I don't think it should be included in the generator, if it is necessary, it is better to define it at the application level.

![image](https://github.com/Carapacik/swagger_parser/assets/34741787/897bed8c-5b0c-4251-8b1a-e9eb77c81da0)

Reply:
Removed.